c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1614,6 +1615,12 @@ class THCCachingAllocator {
     Block* block = device_allocator[device]->malloc(device, size, stream);
     add_allocated_block(block);
     *devPtr = (void*)block->ptr;
+    const auto* interp = c10::impl::CUDATraceTLS::get_trace();
+    if (interp) {
+      interp->trace_cuda_memory_allocation(
+        reinterpret_cast<uintptr_t>(*devPtr)
+      );
+    }
```

Comment:
I was expecting this block to be just next to the call to `cudaMalloc`, but I don't see it. I guess that call happens inside the `device_allocator[device]->malloc(...)`. Should we move this hook within that function, thus closer to the `cudaMalloc` call-site?

The same applies for the hook below, and the `cudaFree` callsite.

Reply:
As discussed, I will investigate this option further.